src/util/rocksdb_util/properties.rs

```diff
@@ -940,4 +953,64 @@ mod tests {
             );
         }
     }
+
+    #[test]
+    fn test_range_properties_with_blob_index() {
+        let cases = [
+            ("a", 0, 1),
+            // handle "a": size(size = 1, offset = 1),keys(1,1)
+            ("b", PROP_SIZE_INDEX_DISTANCE / 8, 1),
+            ("c", PROP_SIZE_INDEX_DISTANCE / 4, 1),
+            ("d", PROP_SIZE_INDEX_DISTANCE / 2, 1),
+            ("e", PROP_SIZE_INDEX_DISTANCE / 8, 1),
+            // handle "e": size(size = DISTANCE + 4, offset = DISTANCE + 5),keys(4,5)
+            ("f", PROP_SIZE_INDEX_DISTANCE / 4, 1),
+            ("g", PROP_SIZE_INDEX_DISTANCE / 2, 1),
+            ("h", PROP_SIZE_INDEX_DISTANCE / 8, 1),
+            ("i", PROP_SIZE_INDEX_DISTANCE / 4, 1),
+            // handle "i": size(size = DISTANCE / 8 * 9 + 4, offset = DISTANCE / 8 * 17 + 9),keys(4,5)
+            ("j", PROP_SIZE_INDEX_DISTANCE / 2, 1),
+            ("k", PROP_SIZE_INDEX_DISTANCE / 2, 1),
+            // handle "k": size(size = DISTANCE + 2, offset = DISTANCE / 8 * 25 + 11),keys(2,11)
+        ];
+
+        let handles = ["a", "e", "i", "k"];
+
+        let mut rng = rand::thread_rng();
+        let mut collector = RangePropertiesCollector::default();
+        let mut extra_value_size: u64 = 0;
+        for &(k, vlen, count) in &cases {
+            if handles.contains(&k) || rng.gen_range(0, 2) == 0 {
```

Comment:
Is it necessary to add randomness here?

Reply:
doesn't harm

Reply:
It harms because the reviewer need to think about it and find out that it is unnecessary.

Reply:
no, it's necessary

Reply:
Can you explain the test? I'm a bit confusing.